src/js/providers/bandwidth-monitor.ts

```diff
@@ -1,14 +1,14 @@
 import { BANDWIDTH_ESTIMATE } from 'events/events';
 import { isValidNumber } from 'utils/underscore';
-import type { DefaultProvider } from 'types/generic.type';
+import type { ProviderWithMixins } from './default';
```

Comment:
This module depends on providers implementing two methods, `trigger` and `getBandwidthEstimate`. It's in src/js/providers so it's only imported by providers when can provide an estimate. It relies in no way on mixins. Based on what you have, it looks like `ImplementedProvider` would be the right type to use here.

Reply:
Nah `trigger` is part of the `Events` mixin but would also be required for external providers so i'm going to update the types holistically here, i'll use the external facing provider interface which should include the trigger method and leave `ImplementedProvider` (potentially to be renamed) as the provider base for our mixin setup internally